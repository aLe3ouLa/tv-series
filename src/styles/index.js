import React from "react";

import { Reset, Typography } from "./defaults";
import { color } from "./variables/colors";
import { spacing } from "./variables/spacings";
import { breakpoint } from "./variables/breakpoints";

export const Defaults = () => (
  <>
    <Reset />
    <Typography />
  </>
);

export const theme = {
  color,
  spacing,
  breakpoint,
};
