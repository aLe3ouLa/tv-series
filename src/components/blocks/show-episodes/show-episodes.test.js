import React from "react";
import ShowEpisodes from "./";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../styles";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Show Episodes component", () => {
  const mockProps = {
    groups: [[{ id: "1", name: "test" }], [{ id: "2", name: "test1" }]],
  };
  afterEach(cleanup);


  it("section for episodes should render two children", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowEpisodes {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const titleWrapper = getByTestId("test-1");
    expect(titleWrapper.children.length).toBe(2);
  });

  it("section for episodes has a season title", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowEpisodes {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const title = getByTestId("title-1");
    expect(title.textContent).toBe("Season 0");
  });

  it("card list has 1 children in season 0", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowEpisodes {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const title = getByTestId("card-list-1");
    expect(title.children.length).toBe(1);
  });
});
