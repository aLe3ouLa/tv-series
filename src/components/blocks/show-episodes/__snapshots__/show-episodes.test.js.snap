// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Episodes component should render shows episode component without crashing 1`] = `
<div>
  <section
    class="sc-AxjAm jHXAgG"
  >
    <h2>
      Episodes
    </h2>
    <div
      data-testid="test-0"
    >
      <h3
        data-testid="title-0"
      >
        Season 
        1
      </h3>
      <ul
        class="sc-AxgMl dqEHLu"
        data-testid="card-list-0"
      >
        <li
          class="sc-AxirZ hiCPGx"
        >
          <a
            class="sc-AxhCb hvQPKY"
            href="/episodes/2"
          >
            <h5
              class="sc-AxiKw kUnrUt"
            >
              test1
            </h5>
          </a>
          <img
            alt="test1"
            src="http://static.tvmaze.com/images/no-img/no-img-landscape-text.png"
          />
          <p
            class="sc-AxhUy ivAIfo"
          >
            <span>
              <strong>
                Episode:
              </strong>
               
              x
            </span>
            <span>
              <strong>
                Airdate:
              </strong>
               
            </span>
          </p>
          <span />
        </li>
      </ul>
    </div>
    <div
      data-testid="test-1"
    >
      <h3
        data-testid="title-1"
      >
        Season 
        0
      </h3>
      <ul
        class="sc-AxgMl dqEHLu"
        data-testid="card-list-1"
      >
        <li
          class="sc-AxirZ hiCPGx"
        >
          <a
            class="sc-AxhCb hvQPKY"
            href="/episodes/1"
          >
            <h5
              class="sc-AxiKw kUnrUt"
            >
              test
            </h5>
          </a>
          <img
            alt="test"
            src="http://static.tvmaze.com/images/no-img/no-img-landscape-text.png"
          />
          <p
            class="sc-AxhUy ivAIfo"
          >
            <span>
              <strong>
                Episode:
              </strong>
               
              x
            </span>
            <span>
              <strong>
                Airdate:
              </strong>
               
            </span>
          </p>
          <span />
        </li>
      </ul>
    </div>
  </section>
</div>
`;
